lib/std/mem.zig

```diff
@@ -175,6 +175,9 @@ pub const Allocator = struct {
 
         const old_byte_slice = @sliceToBytes(old_mem);
         const byte_count = math.mul(usize, @sizeOf(T), new_n) catch return Error.OutOfMemory;
+        if (new_n < old_mem.len) {
+            @memset(old_byte_slice.ptr + byte_count, undefined, old_byte_slice.len - byte_count);
```

Comment:
Thinking more... this is incorrect isn't it? If the `realloc` fails then the memory should be left untouched. Will remove from this PR. Sad :(